orttraining/orttraining/test/python/orttraining_test_onnxblock.py

```diff
@@ -816,3 +817,57 @@ def mse_loss(prediction, target):
         # assert all the gradients are close
         for ort_grad, pt_param in zip(ort_outs[0], pt_model.parameters()):
             assert np.allclose(ort_grad, _to_numpy(pt_param.grad))
+
+
+def test_eval_model_has_no_training_mode_dropout():
+    class DropoutModel(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.dropout = torch.nn.Dropout(p=0.5)
+
+        def forward(self, x):
+            return self.dropout(x)
+
+    model = DropoutModel()
+    onnx_model = _get_onnx_model(model, (torch.randn(1, 3, 224, 224),))
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        artifacts.generate_artifacts(onnx_model, loss=artifacts.LossType.CrossEntropyLoss, artifact_directory=temp_dir)
+
+        eval_model = onnx.load(os.path.join(temp_dir, "eval_model.onnx"))
+
+        flag = False
+        for node in eval_model.graph.node:
+            if node.op_type == "Dropout":
+                assert not node.input[2]
+                flag = True
+
+        assert flag
+
+
+def test_eval_model_has_no_training_mode_batchnorm():
+    class BatchNormModel(torch.nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.batchnorm = torch.nn.BatchNorm2d(100)
+
+        def forward(self, x):
+            return self.batchnorm(x)
+
+    model = BatchNormModel()
+    onnx_model = _get_onnx_model(model, (torch.randn(20, 100, 35, 45),))
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        artifacts.generate_artifacts(onnx_model, loss=artifacts.LossType.CrossEntropyLoss, artifact_directory=temp_dir)
+
+        eval_model = onnx.load(os.path.join(temp_dir, "eval_model.onnx"))
+
+        flag = False
+        for node in eval_model.graph.node:
+            if node.op_type == "BatchNormalization":
+                for attr in node.attribute:
+                    if attr.name == "training_mode":
```

Comment:
A related question: BatchNorm 14 starts having `training_mode`, do we need let user to use OPSET>=14 to export the inference model? Have we differentiate the logic for generating training graph (BatchNormInternal) and eval graph (BatchNorm) currently?

Reply:
Training graph = gradient_graph(training_optimizer(stack(forward_graph, loss)))
eval graph = stack(forward_graph, loss).

So, eval model will never have batchnorminternal unless that base model provided by the user has batchnorminternal.